src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptionhandlers/TaskErrorHandler.java

```diff
@@ -71,4 +76,10 @@ private void registerEvent(EventRelatedThrowable exception, String reason) {
             processEvent.getEvent()
         );
     }
+
+    private void incrementUploadFailureCount(DocUploadFailureGenericException exc) {
+        Envelope envelope = exc.getEnvelope();
+        envelope.setUploadFailureCount(envelope.getUploadFailureCount() + 1);
```

Comment:
Maybe worth having `public void incrementFailureCount()` method inside the `Envelope`?

Reply:
Considering we're leaning toward replacing jpa with jdbc, I'd rather keep envelope model as 'thin' as possible.

Reply:
let's do this! i mean the replacing 😉